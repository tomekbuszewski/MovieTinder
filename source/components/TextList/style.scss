@import '../App/globals';

$Element: '.TextList';

#{$Element} {
  color: map_get($colors, dark);
  font-size: map_get($fonts, mid);

  &--Small {
    display: flex;
    list-style: none;

    #{$Element}__Item {
      margin-right: 20px;
    }
  }
}
