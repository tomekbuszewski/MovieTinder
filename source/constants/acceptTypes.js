export const MOVIE_YES = 'MOVIE_YES';
export const MOVIE_NO = 'MOVIE_NO';
