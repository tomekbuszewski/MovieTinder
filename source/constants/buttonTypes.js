export const BUTTON_ACCEPT = 'BUTTON_ACCEPT';
export const BUTTON_DECLINE = 'BUTTON_DECLINE';
