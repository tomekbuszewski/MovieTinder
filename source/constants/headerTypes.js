export const LARGE_HEADER = 'LARGE';
export const SMALL_HEADER = 'SMALL';
