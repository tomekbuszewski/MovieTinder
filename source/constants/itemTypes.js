export const SMALL_ITEM = 'SMALL';
export const LARGE_ITEM = 'LARGE';
export const NORMAL_ITEM = 'NORMAL';
