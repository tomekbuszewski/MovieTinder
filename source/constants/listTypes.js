export const LARGE_LIST = 'LARGE_LIST';
export const SMALL_LIST = 'SMALL_LIST';
