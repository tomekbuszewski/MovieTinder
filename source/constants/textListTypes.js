export const SMALL_TEXT_LIST = 'SMALL';
export const NORMAL_TEXT_LIST = 'NORMAL';
