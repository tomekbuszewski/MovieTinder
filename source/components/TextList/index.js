import React from 'react';
import PropTypes from 'prop-types';
import { NORMAL_TEXT_LIST } from '../../constants/textListTypes';

import style from './style.scss';

import c from '../../services/classNameBuilder';

// @flow
const TextList = ({ items, additionalClassNames, type }: { items: Array, additionalClassNames: string, type: string }) => {
  const mainClassName = type === NORMAL_TEXT_LIST ? style['TextList'] : `${style['TextList']} ${c(style, { b: 'TextList', m: 'Small' })}`;

  return <ul className={`${mainClassName} ${additionalClassNames}`}>
    {items.map((item, i) => <li key={i} className={c(style, { b: 'TextList', e: 'Item' })} dangerouslySetInnerHTML={{ __html: item }} />)}
  </ul>;
};

TextList.propTypes = {
  items: PropTypes.array,
  additionalClassNames: PropTypes.string,
  type: PropTypes.string
};

TextList.defaultProps = {
  items: [],
  additionalClassNames: '',
  type: NORMAL_TEXT_LIST
};

export default TextList;
