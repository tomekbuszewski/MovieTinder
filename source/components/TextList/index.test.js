import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import TextList from './index';
import { SMALL_TEXT_LIST } from '../../constants/textListTypes';

describe('Text List tests', () => {
  it ('default list view should match snapshot', () => {
    const component = renderer.create(<TextList items={[ 'one', 'two', ]} />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it ('small list view should match snapshot', () => {
    const component = renderer.create(<TextList items={[ 'one', 'two', ]} type={SMALL_TEXT_LIST} />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it ('should make `li` elements from given items', () => {
    const component = mount(<TextList items={[ 'one', 'two', ]} />);

    expect(component.find('li')).toHaveLength(2);
  })
});
