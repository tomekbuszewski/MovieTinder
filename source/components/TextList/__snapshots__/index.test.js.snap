// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text List tests default list view should match snapshot 1`] = `
<ul
  className="TextList "
>
  <li
    className=""
    dangerouslySetInnerHTML={
      Object {
        "__html": "one",
      }
    }
  />
  <li
    className=""
    dangerouslySetInnerHTML={
      Object {
        "__html": "two",
      }
    }
  />
</ul>
`;

exports[`Text List tests small list view should match snapshot 1`] = `
<ul
  className="TextList  "
>
  <li
    className=""
    dangerouslySetInnerHTML={
      Object {
        "__html": "one",
      }
    }
  />
  <li
    className=""
    dangerouslySetInnerHTML={
      Object {
        "__html": "two",
      }
    }
  />
</ul>
`;
